dev/com.ibm.ws.classloader.context/src/com/ibm/ws/classloader/context/internal/ClassloaderContextImpl.java

```diff
@@ -144,14 +148,27 @@ public void taskStopping() {
 
     /**
      * Sets the provided classloader on the current thread.
-     * 
+     *
      * @param cl The clasloader to be set.
      */
     private void setCL(final ClassLoader cl) {
         PrivilegedAction<Object> action = new PrivilegedAction<Object>() {
             @Override
             public Object run() {
-                Thread.currentThread().setContextClassLoader(cl);
+                final Thread t = Thread.currentThread();
+                try {
+                    t.setContextClassLoader(cl);
+                } catch (SecurityException e) {
+                    // If this work happens to run on an java.util.concurrent.ForkJoinWorkerThread$InnocuousForkJoinWorkerThread,
+                    // setting the ClassLoader may be rejected. If this happens, give a decent error message.
+                    if (t instanceof ForkJoinWorkerThread && "InnocuousForkJoinWorkerThreadGroup".equals(t.getThreadGroup().getName())) {
+                        String msg = Tr.formatMessage(tc, "cannot.apply.classloader.context");
+                        Tr.error(tc, msg);
```

Comment:
Does this show up properly in logs?  I suspect Tr.error will complain that it does not find a key matching the translated message.  That said, I think we should remove the Tr.error for this.  The exception already flows back to the application and is covered by FFDC.

Reply:
yep, it does.  I will remove the `Tr.error` though